### PR TITLE
[cu-30xvvc2, cu-30xvvc7][Pablo Refactor] Adding new function signatures and types

### PR DESCRIPTION
**Pull request by vimukthi-git**
## Description
- Introducing new types and function signatures for `Amm` trait
- Handling fall out in Pablo
- Handling fall out in Dex-router

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-30xvvc7 5.3) 2. Implement Algorithm: Amm::currency_pair → Amm::assets.](https://app.clickup.com/t/30xvvc7)

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-30xvvc2 5.3) 1. Implement Refactoring: CurrencyPair Usage.](https://app.clickup.com/t/30xvvc2)

File: code/parachain/frame/composable-traits/src/dex.rs [hunk 1]
```diff
@@ -11,6 +11,44 @@ use serde::{Deserialize, Serialize};
 use sp_runtime::{BoundedBTreeMap, DispatchError, Permill};
 use sp_std::{collections::btree_map::BTreeMap, fmt::Debug, ops::Mul, vec::Vec};
 
+/// Specifies and amount together with the asset ID of the amount.
```
Comment (vimukthi-git) [inline code/parachain/frame/composable-traits/src/dex.rs:14]: Main change is in this file, everything else is the fallout of it

**Review by PoisonPhang — APPROVED** [review]

**Review by dzmitry-lahoda — APPROVED** [review]

File: code/parachain/frame/composable-traits/src/dex.rs [hunk 2]
```diff
@@ -11,6 +11,44 @@ use serde::{Deserialize, Serialize};
...
+#[derive(
```
Comment (dzmitry-lahoda) [inline code/parachain/frame/composable-traits/src/dex.rs:15]: 🥇 Happy birthday!!!

Default asset in Composable is invalid. You cannot have 0 asset. So zero amount also kind of also not type (zero amount is like no operation or None). So may be if it compiles `Default` can be removed. So not sure.

**Comment by vimukthi-git** [comment]
@mergifyio refresh

**Pull request merged**